README.md

```diff
@@ -1,42 +1,422 @@
-    <img src="resources/images/flowchart.svg" alt="Scheme"  width="910" height="459">
+    <img src="resources/images/datastructure.svg" alt="Scheme"  width="910" height="459">
   </a>
   <p align="center">
-    Description of PASEOS workflow on an individual device
+    Description of PASEOS data structure
   </p>
 </p>
 
-## Visualising PASEOS
-Navigate to paseos/visualization to find a jupyter notebook containing examples of how to visualize PASEOS.
-Visualization can be done in interactive mode or as an animation that is saved to disc.
-In the figure below, Earth is visualized in the center as a blue sphere with different spacecraft in orbit.
-Each spacecraft has a name and if provided, a battery level and a communications device.
-The local device is illustrated with white text.
-In the upper-right corner, the status of the communication link between each spacecraft is shown.
-Finally, the time in the lower left and lower right corners correspond to the epoch and the PASEOS local simulation time.
-
 <p align="center">
   <a href="https://github.com/aidotse/PASEOS/">
-    <img src="resources/images/animation.png" alt="Scheme"  width="900" height="500">
+    <img src="resources/images/flowchart.svg" alt="Scheme"  width="910" height="459">
   </a>
   <p align="center">
-    Snapshot of PASEOS visualization
+    Description of PASEOS workflow on an individual device
   </p>
-</p>
\ No newline at end of file
+</p>
+
+## Glossary
+* ### Activity
+  An activity is the abstraction that PASEOS uses to keep track of specific actions performed by a [SpacecraftActor](#spacecraftactor) upon a request from the user. To register an activity, an user shall first create an [activity function](#activity-function), which describes the operation to be performed, and provide information on the power-consumption due to the activity execution. <br>PASEOSis responsible for the execution of the activity and for updating the system status depending on the effects of the activity (e.g., by discharging they satellite battery).<br>
+  When registering an activity, the user can specify a [constraint-function](#constraint-function) to specify constraints to be met during the execution of the activity and an an [on termination](#on-termination) function to specify additional operations to performed by PASEOS on termination of the activity function.
+
+
+* ### Activity function
+  User-defined function emulating any operation to be executed in a PASEOS a [SpacecraftActor](#spacecraftactor). Activity functions are necessayr to register [activities](#activity). Activity functions might include data transmission, house-keeping operations, onboard data acquisition and processing, and others.
+
+* ### Actor
+  Since PASEOS is fully-decentralised, each node of a PASEOS constellation shall run an instance of PASEOS modelling all the nodes of  that constellation.  The abstraction of a constellation node inside a PASEOS instace is a PASEOS `actor`. 
```

Comment:
```suggestion
  Since PASEOS is fully-decentralised, each node of a PASEOS constellation shall run an instance of PASEOS modelling all the nodes of  that constellation.  The abstraction of a constellation node inside a PASEOS instance is a PASEOS `actor`. 
```